### PR TITLE
Group4 rpc impl

### PR DESCRIPTION
**Pull request by Swyrough**
Implementation/infrastructure for some client-namenode rpc calls

### CODE REVIEW

File: rpcserver/rpcserver.cc [hunk 1]
```diff
@@ -131,6 +132,20 @@ void RPCServer::handle_rpc(tcp::socket sock) {
     }
 }
 
+/**
+ * Register all the rpc functions handlers (currently just client to namenode) 
+ */
+void RPCServer::register_handlers() {
+	using namespace client_namenode_translator;
+
+	register_handler("getFileInfo",	 		ClientNamenodeTranslator::getFileInfo);
```
Comment (efrederic) [inline rpcserver/rpcserver.cc]: The server shouldn't know what handlers are going to be registered. We want to move this over to the rice-namenode main.cc

Comment (efrederic) [reply rpcserver/rpcserver.cc]: Everything else in this pull request looks good to me btw

**Comment by pelmers** [comment]
I think handler registering should be done in namenode rather than directly in rpcserver, since both datanode and namenode will use rpcserver.

**Comment by pelmers** [comment]
oops, already mentioned by ethan

File: README.md [hunk 2]
```diff
@@ -37,4 +37,8 @@ cmake ..
 make
 ```
 You will see a sample executable placed in `build/rice-namenode/namenode.` The
-compiled protocols are in `build/proto`.
+compiled protocols are in `build/proto`
+
+# Testing
+Run the namenode executable. Then run something like "hdfs dfs -fs hdfs://localhost:5351/ -mkdir foo"
```
Comment (pelmers) [inline README.md]: Let's put the command in backticks to make it stand out more in the web view. `hdfs dfs -fs hdfs://localhost:5351/ -mkdir foo`

File: rice-namenode/ClientNamenodeProtocolImpl.cc [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	AppendResponseProto res;
+	// TODO We don't support this operation, so we need to return some
+	// kind of failure status. I've looked around and I'm not sure 
+	// how to do this since this message only contains an optional
+	// LocatedBlockProto. No LocatedBlockProto might be failure
+	return Serialize(&out, res);
+}
+
+// TODO delete is a keyword in C++. Come up with a better replacement name than
+// deleteCmd
```
Comment (pelmers) [inline rice-namenode/ClientNamenodeProtocolImpl.cc]: how about 'destroy'

File: rpcserver/rpcserver.cc [hunk 4]
```diff
@@ -73,7 +73,7 @@ bool RPCServer::receive_prelude(tcp::socket& sock) {
 
 /**
  * Handle a single RPC connection from Hadoop's ProtobufRpcEngine.
- */
+ */
```
Comment (pelmers) [inline rpcserver/rpcserver.cc]: delete space at end of line

File: rpcserver/rpcserver.h [hunk 5]
```diff
@@ -3,18 +3,21 @@
...
+#pragma once
+
 using asio::ip::tcp;
 
 class RPCServer {
     public:
         RPCServer(int port);
         void serve(asio::io_service& io_service);
-        void register_handler(std::string key, std::function<std::string(std::string)> handler);
+		void register_handlers();
```
Comment (pelmers) [inline rpcserver/rpcserver.h]: also i don't see us ever using this. or defining it. forgot to delete?

**Review by pelmers — APPROVED** [review]
r+ looks good

When this lands on develop can you make an issue that we need to handle configuration?

**Comment by pelmers** [comment]
Oh right. One more thing. I hesitate to print the bytes of protobuf messages directly. I suggest to use DebugString() method instead to get a nice representation.

**Comment by jfking** [comment]
Hey @Swyrough remember to move this along on the project board

**Pull request closed without merging**